File: frontend/lib/models/FichaMedica.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IFichaMedica extends Document {
  patientId: string;
  historicoFamiliar?: string;
  cirurgias?: string;
  fuma?: boolean;
  bebe?: boolean;
  habitosGerais?: string;
  observacoes?: string;
  updatedAt: Date;
}

const FichaMedicaSchema: Schema = new Schema({
  patientId: { type: String, required: true, unique: true },
  historicoFamiliar: { type: String },
  cirurgias: { type: String },
  fuma: { type: Boolean, default: false },
  bebe: { type: Boolean, default: false },
  habitosGerais: { type: String },
  observacoes: { type: String },
  updatedAt: { type: Date, default: Date.now }
}, {
  collection: 'FichaMedicas'
});

export default mongoose.models.FichaMedica || mongoose.model<IFichaMedica>('FichaMedica', FichaMedicaSchema);

File: backend/SaudeMemora.Domain/Entities/FichaMedica.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.ComponentModel.DataAnnotations;

namespace SaudeMemora.Domain.Entities;

public class FichaMedica
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    [BsonElement("patientId")]
    [Required]
    public string PatientId { get; set; } = string.Empty;

    [BsonElement("historicoFamiliar")]
    public string HistoricoFamiliar { get; set; } = string.Empty;

    [BsonElement("cirurgias")]
    public string Cirurgias { get; set; } = string.Empty;

    [BsonElement("fuma")]
    public bool Fuma { get; set; } = false;

    [BsonElement("bebe")]
    public bool Bebe { get; set; } = false;

    [BsonElement("habitosGerais")]
    public string HabitosGerais { get; set; } = string.Empty;

    [BsonElement("observacoes")]
    public string Observacoes { get; set; } = string.Empty;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("condicoes")]
    public List<CondicaoMedica> Condicoes { get; set; } = new();

    [BsonElement("outrasDoencas")]
    public string OutrasDoencas { get; set; } = string.Empty;

    [BsonElement("tipoSanguineo")]
    public string? TipoSanguineo { get; set; }

    [BsonElement("doadorOrgaos")]
    public bool DoadorOrgaos { get; set; } = false;

    [BsonElement("alergias")]
    public List<string> Alergias { get; set; } = new();

    [BsonElement("doencasCronicas")]
    public List<string> DoencasCronicas { get; set; } = new();
}

public class CondicaoMedica
{
    [BsonElement("nome")]
    public string Nome { get; set; } = string.Empty;

    [BsonElement("tem")]
    public bool Tem { get; set; } = false;

    [BsonElement("detalhes")]
    public string Detalhes { get; set; } = string.Empty;
}
